packages/devtools_server/lib/src/file_system.dart

```diff
@@ -43,8 +43,13 @@ class LocalFileSystem {
 
   /// Returns a DevTools file from the given path.
   ///
-  /// Only files withing ~/.devtools/ can be accessed.
+  /// Only files within ~/.devtools/ can be accessed.
   static File devToolsFileFromPath(String pathFromDevToolsDir) {
+    if (pathFromDevToolsDir.contains('../')) {
+      // The passed in path should not be able to walk up the directory tree
+      // outside of the ~/.dart-devtools/ directory.
```

Comment:
isn't this flutter-devtools not dart-devtools?

Reply:
done